android/src/main/java/com/lykhonis/flutterbilling/BillingPlugin.java

```diff
@@ -190,8 +199,12 @@ public void failed() {
 
         final List<String> identifiers = new ArrayList<>(purchases.size());
 
+        System.out.println("fetchPurchases, trying to consume:");
+
         for (Purchase purchase : purchases) {
             identifiers.add(purchase.getSku());
+            System.out.println(purchase.getPurchaseToken());
+            billingClient.consumeAsync(purchase.getPurchaseToken(), onConsumeListener);
```

Comment:
I did not look at iOS yet, but this should be explicit, so a client (Dart code) can call to consume identifier when needed.